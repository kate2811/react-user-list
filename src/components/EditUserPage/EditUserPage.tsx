import React, { useMemo } from 'react'
import UserForm from '../UserForm'
import PageLayout from '../PageLayout'
import { User } from '../../modules/core/types'

type Props = {
  editUser: any
  user: User
}

const EditUserPage: React.FC<Props> = ({ editUser, user }) => {
  const breadcrumbs = useMemo(() => {
    return [{ title: 'Edit user' }, { title: `${user.name} ${user.surname}` }]
  }, [user])

  return (
    <PageLayout title={'Edit user'} breadcrumbs={breadcrumbs}>
      <UserForm onSave={editUser} user={user} />
    </PageLayout>
  )
}

export default EditUserPage
