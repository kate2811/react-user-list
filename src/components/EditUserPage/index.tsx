import React from 'react'
import { useEditUser, useUserById } from '../../modules/core/hooks'
import EditUserPage from './EditUserPage'
import { useParams } from 'react-router-dom'

export default function() {
  const editUser = useEditUser()
  const { id } = useParams()
  if (!id) {
    throw new Error('no user id is provided')
  }
  const user = useUserById(id)
  if (!user) {
    throw new Error('user is not found')
  }

  return <EditUserPage editUser={editUser} user={user} />
}
