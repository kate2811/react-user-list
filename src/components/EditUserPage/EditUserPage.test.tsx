import React from 'react'
import { render } from '@testing-library/react'
import EditUserPage from './EditUserPage'
import { MemoryRouter } from 'react-router-dom'

const onClick = jest.fn()

describe('Edit User page', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('Edit user page is rendered', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <EditUserPage
          editUser={onClick}
          user={{ name: 'mockename', surname: 'surname', age: 1, email: 'kk@kk.r', id: '1' }}
        />
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('Is save button disabled', () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <EditUserPage
          editUser={onClick}
          user={{ name: 'mockename', surname: 'surname', age: 1, email: 'kk@kk.r', id: '1' }}
        />
      </MemoryRouter>
    )
    expect(getByTestId('success-button')).toBeDisabled()
  })
})
