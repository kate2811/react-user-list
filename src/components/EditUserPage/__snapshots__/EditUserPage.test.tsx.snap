// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit User page Edit user page is rendered 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-dark bg-primary justify-content-start"
    >
      <span
        class="navbar-brand mb-0 h1"
      >
        UserList
      </span>
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-link active"
          href="/"
        >
          Home
        </a>
      </div>
    </nav>
    <nav
      aria-label="breadcrumb"
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          class="breadcrumb-item active"
          data-testid="path item"
        >
          Edit user
        </li>
        <li
          aria-current="page"
          class="breadcrumb-item active"
          data-testid="path item"
        >
          mockename surname
        </li>
      </ol>
    </nav>
    <div
      class="container mt-4"
    >
      <h1
        class="mb-3"
      >
        Edit user
      </h1>
      <div>
        <form
          action="#"
          data-testid="form"
        >
          <div
            class="form-group"
          >
            <label
              for="name"
            >
              Name
            </label>
            <input
              class="form-control form-control"
              id="name"
              name="name"
              type="text"
              value="mockename"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="surname"
            >
              Surname
            </label>
            <input
              class="form-control form-control"
              id="surname"
              name="surname"
              type="text"
              value="surname"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="email"
            >
              Email
            </label>
            <input
              class="form-control form-control"
              id="email"
              name="email"
              type="text"
              value="kk@kk.r"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="age"
            >
              Age
            </label>
            <input
              class="form-control form-control"
              id="age"
              name="age"
              type="number"
              value="1"
            />
          </div>
          <div
            class="d-flex justify-content-between mt-4"
          >
            <button
              class="btn btn-primary w-25"
              data-testid="success-button"
              disabled=""
              type="submit"
            >
              Save
            </button>
            <a
              class="btn btn-dark w-25"
              href="/"
            >
              Go back
            </a>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
